### PR TITLE
change to map based config, match wildcard path

### PR DESCRIPTION
**Pull request by mikehelmick**
## Proposed Changes

* change to map based config for redirect service
* match wildcard

**Release Note**

```release-note
NONE
```

/assign @icco 
/assign @sethvargo

### CODE REVIEW

File: pkg/config/redirect_config.go [hunk 1]
```diff
@@ -70,13 +63,9 @@ func (c *RedirectConfig) ObservabilityExporterConfig() *observability.Config {
...
-	if len(c.Hostnames) != len(c.RegionCodes) {
-		return nil, fmt.Errorf("HOSTNAME and REGION_CODE must be lists of the same length")
-	}
-
-	hostnameToRegion := make(map[string]string, len(c.Hostnames))
-	for i, prefix := range c.Hostnames {
-		hostnameToRegion[strings.ToLower(prefix)] = strings.ToUpper(c.RegionCodes[i])
+	hostnameToRegion := make(map[string]string, len(c.HostnameConfig))
+	for hostname, region := range c.HostnameConfig {
+		hostnameToRegion[strings.ToLower(hostname)] = strings.ToUpper(region)
```
Comment (icco) [inline pkg/config/redirect_config.go:68]: Log an error if a region is empty or invalid?

File: cmd/enx-redirect/main.go [hunk 2]
```diff
@@ -95,7 +95,9 @@ func realMain(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	r.Handle("/", redirectController.HandleIndex()).Methods("GET")
+	r.PathPrefix("/").Handler(redirectController.HandleIndex()).Methods("GET")
+	//s.Handle("/", redirectController.HandleIndex()).Methods("GET")
+	//r.Handle("/v", redirectController.HandleIndex()).Methods("GET")
```
Comment (icco) [inline cmd/enx-redirect/main.go:100]: Delete unused

File: pkg/config/redirect_config.go [hunk 3]
```diff
@@ -70,13 +63,9 @@ func (c *RedirectConfig) ObservabilityExporterConfig() *observability.Config {
 }
 
 func (c *RedirectConfig) HostnameToRegion() (map[string]string, error) {
```
Comment (sethvargo) [inline pkg/config/redirect_config.go:65]: Document that this is lowercase? Or document that `HostnameConfig` is case-insensitive.

**Review by sethvargo — APPROVED** [review]
lgtm

**Pull request merged**